probatus/feature_elimination/feature_elimination.py

```diff
@@ -695,3 +680,259 @@ def plot(self, show=True, **figure_kwargs):
+        The main difference is removing the lowest importance features based on SHAP features importance. It also
+        supports the use of sklearn compatible search CV for hyperparameter optimization e.g.
+        [GridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LassoCV.html),
+        [RandomizedSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html?highlight=randomized#sklearn.model_selection.RandomizedSearchCV), or
+        [BayesSearchCV](https://scikit-optimize.github.io/stable/modules/generated/skopt.BayesSearchCV.html), which
+        needs to be passed as the `clf`. Thanks to this you can perform hyperparameter optimization at each step of
+        the feature elimination. Lastly, it supports categorical features (object and category dtype) and missing values
+        in the data, as long as the model supports them.
+
+    We recommend using [LGBMClassifier](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMClassifier.html),
+        because by default it handles missing values and categorical features. In case of other models, make sure to
+        handle these issues for your dataset and consider impact it might have on features importance.
+
+
+    Example:
+    ```python
+    from lightgbm import LGBMClassifier
+    import numpy as np
+    import pandas as pd
+    from probatus.feature_elimination import EarlyStoppingShapRFECV
+    from sklearn.datasets import make_classification
+    from sklearn.model_selection import train_test_split
+    from sklearn.ensemble import RandomForestClassifier
+    from sklearn.model_selection import RandomizedSearchCV
+
+    feature_names = [
+        'f1', 'f2', 'f3', 'f4', 'f5', 'f6', 'f7',
+        'f8', 'f9', 'f10', 'f11', 'f12', 'f13',
+        'f14', 'f15', 'f16', 'f17', 'f18', 'f19', 'f20']
+
+    # Prepare two samples
+    X, y = make_classification(n_samples=200, class_sep=0.05, n_informative=6, n_features=20,
+                               random_state=0, n_redundant=10, n_clusters_per_class=1)
+    X = pd.DataFrame(X, columns=feature_names)
+
+
+    # Prepare model and parameter search space
+    clf = LGBMClassifier(n_estimators=200)
+
+    param_grid = {
+        "max_depth": [2, 4, 8, 16],
+        "num_leaves": [3, 5, 7, 10],
+    }
+    search = RandomizedSearchCV(clf, param_grid)
+
+
+    # Run feature elimination
+    shap_elimination = EarlyStoppingShapRFECV(
+        clf=search, step=0.2, cv=10, scoring='roc_auc', early_stopping_rounds=10, n_jobs=3)
+    report = shap_elimination.fit_compute(X, y)
+
+    # Make plots
+    performance_plot = shap_elimination.plot()
+
+    # Get final feature set
+    final_features_set = shap_elimination.get_reduced_features_set(num_features=3)
+    ```
+    <img src="../img/earlystoppingshaprfecv.png" width="500" />
+
+    """  # noqa
+
+    def __init__(
+        self,
+        clf,
+        early_stopping_rounds=None,
```

Comment:
maybe we can set it to something that makes sense, and allows users to run it straight away without error?

How about `5`?